### PR TITLE
KAFKA-18282: Add StreamsGroupHeartbeat RPC definitions

### PR DESCRIPTION
**Pull request by lucasbru**
The StreamsGroupHeartbeat API is the new core API used by streams application to form a group. The API allows members to initialize a topology, advertise their state, and their owned tasks. The group coordinator uses it to assign/revoke tasks to/from members. This API is also used as a liveness check.

This change adds the JSON definition of the RPC, as defined in KIP-1071.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/requests/StreamsGroupHeartbeatResponse.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * - {@link Errors#COORDINATOR_LOAD_IN_PROGRESS}
+ * - {@link Errors#INVALID_REQUEST}
+ * - {@link Errors#UNKNOWN_MEMBER_ID}
+ * - {@link Errors#FENCED_MEMBER_EPOCH}
+ * - {@link Errors#UNSUPPORTED_ASSIGNOR}
+ * - {@link Errors#UNRELEASED_INSTANCE_ID}
+ * - {@link Errors#GROUP_MAX_SIZE_REACHED}
+ * - {@link Errors#STREAMS_INVALID_TOPOLOGY}
+ * - {@link Errors#STREAMS_INVALID_TOPOLOGY_EPOCH}
+ * - {@link Errors#STREAMS_TOPOLOGY_FENCED}
```
Comment (cadonna) [inline clients/src/main/java/org/apache/kafka/common/requests/StreamsGroupHeartbeatResponse.java:45]: In the KIP, we also have the following errors that are not in this list:
- `GROUP_ID_NOT_FOUND`
- `TOPIC_AUTHORIZATION_FAILED`
- `CLUSTER_AUTHORIZATION_FAILED`

In this list, the following errors are not in the KIP:
- `UNSUPPORTED_ASSIGNOR`

**Review by cadonna — COMMENTED** [review]
@lucasbru Thanks for the PR!

Here my feedback!

File: clients/src/main/java/org/apache/kafka/common/requests/StreamsGroupHeartbeatResponse.java (hunk 1 above)
Comment (lucasbru) [reply clients/src/main/java/org/apache/kafka/common/requests/StreamsGroupHeartbeatResponse.java:45]: Done

**Review by cadonna — APPROVED** [review]
Thanks for the updates, @lucasbru!

LGTM!

**Pull request merged**